poi/src/main/java/org/apache/poi/hssf/model/DrawingManager2.java

```diff
@@ -19,21 +19,156 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
```

Comment:
To be clear, the first commit in the description, which is cherry-picked, is from this release, the smartcrypt release from october. I will add phrasing to make that more clear https://github.com/apache/poi/compare/trunk...pkware:poi:smartcryptOctober2022 - this change needed to make it back to `trunk` so it will be the first commit in this PR, but im probably not going to address feedback relating to it.

Reply:
this was code already in for a smartcrypt release in October? why was no PR made for it then?

There are alot of small changes in this code that dont really align it with core-libraries code standards (or general coding standards) we have for variable names/etc so I would like that to be consistent with everything else.

I am okay with this being a separate commit within this PR, but it definitely should not be squashed if this was cherry picked.

If we are squashing this commit then id like to see the same coding standards applied to this commit as the other commits have (variable naming, etc).

Reply:
also was this commit cherryPicked from code changes we made to POI, or from someone else working on POI?

I would only advocate for changing anything if its the former. 

If its the latter, I would definitely indicate in the commit message who the original author was- because git blame will lead all these changes back to you (if we decide to change this code again in the future)

Reply:
> this was code already in for a smartcrypt release in October? why was no PR made for it then?

i dont know but i pinged marius and it _should_ make it back to the trunk, so here we are

>  but it definitely should not be squashed if this was cherry picked.

i definitely dont intend to squash these commits, ill squash the <path> commits together, and the two commits for image/path will be pushed. idk if the way i was addressing the PR feedback implies that it was all going to get squashed together, but i dont plan to do that

> this commit cherry picked from code changes we/marius made to POI

marius made these changes

Reply:
hmm if @MariusVolkhart  made these changes directly then im concerned- he would definitely be calling out anyone else for the naming conventions used in these files (and probably have a separate commit for changing them, before any actual code work)

It feels like this commit was scratch work and not really ready for any PR at the time, I would like it to be reworked to made PR worthy if that's the case (if marius really wrote this code)

Reply:
sure, i wasnt really asking about that part though - i more meant for the questions you asked in reply to the comment i left with the link. I thought the link would have answered the questions you asked already, once i clarified my intention to preserve that commit's history, so im just wondering how i can communicate this kind of thing better next time.

Reply:
oh i see what you mean as far as who authored the commit/code originally (it being marius and not an external dev)

I just wanted to emphasize if we do commit this commit as is with no changes (which id rather not) that the commit message should refer back to marius as the author.

Reply:
right on, ill probably pull the image changes into a separate PR if we dont just want them as-is

Reply:
im good with that